internal/auth/auth.go

```diff
@@ -54,6 +54,34 @@ var createAccount = func(account *data.Account) error {
 	return data.CreateAccount(account)
 }
 
+// DeleteAccount takes the specified credentials and soft-deletes a record in
+// the database, returning any errors encountered.
+func DeleteAccount(username string) error {
+	return softDeleteAccount(username)
+}
+
+var softDeleteAccount = func(username string) error {
```

Comment:
Why this convention, instead of placing this in the `DeleteAccount` function?

Reply:
Check out `auth_test` - it makes it possible to fake the actual ORM interaction in the test so that this doesn't need a bunch of database setup